File: app/src/main/java/com/funckyhacker/fileexplorer/util/PageManger.java
package com.funckyhacker.fileexplorer.util;

import java.util.ArrayDeque;
import java.util.Deque;
import javax.inject.Inject;
import javax.inject.Singleton;
import timber.log.Timber;

@Singleton
public class PageManger {

  private Deque<String> stack;

  @Inject
  public PageManger() {
    stack = new ArrayDeque<>();
  }

  public void push(String path) {
    Timber.i("Pushed Item: %s", path);
    stack.offerFirst(path);
  }

  public String pop() {
    String path = stack.poll();
    Timber.i("Popped Item: %s", path);
    return path;
  }

  public void clearAll() {
    stack.clear();
  }

  public int size() {
    return stack.size();
  }
}

File: app/src/main/java/com/funckyhacker/fileexplorer/util/PageManger.kt
package com.funckyhacker.fileexplorer.util

import timber.log.Timber
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PageManger @Inject
constructor() {

    private val stack: Deque<String>

    init {
        stack = ArrayDeque()
    }

    fun push(path: String) {
        Timber.i("Pushed Item: %s", path)
        stack.offerFirst(path)
    }

    fun pop(): String {
        val path = stack.poll()
        Timber.i("Popped Item: %s", path)
        return path
    }

    fun clearAll() {
        stack.clear()
    }

    fun size(): Int {
        return stack.size
    }
}
